tensorboard/plugins/hparams/summary_proto.proto

```diff
@@ -0,0 +1,40 @@
+/* Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+// Defines protos for storing experiment data inside Summary tags.
```

Comment:
Let's name this file plugin_data.proto to be consistent with other plugins:
https://github.com/tensorflow/tensorboard/blob/1.6.0/tensorboard/plugins/scalar/plugin_data.proto
https://github.com/tensorflow/tensorboard/blob/1.6.0/tensorboard/plugins/text/plugin_data.proto

Like the other plugins I think it'd probably be useful to have a wrapper HParamsPluginData message that we can store a version number on, so that we have the ability to special-case different versions should we need to make breaking schema changes.  It could then have a oneof field that can store an Experiment, SessionStartSummary, or SessionEndSummary as needed.